junit-jupiter-api/src/main/java/org/junit/jupiter/api/ClassOrderer.java

```diff
@@ -0,0 +1,239 @@
+/*
+ * Copyright 2015-2020 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * https://www.eclipse.org/legal/epl-v20.html
+ */
+
+package org.junit.jupiter.api;
+
+import static java.util.Comparator.comparingInt;
+import static org.apiguardian.api.API.Status.EXPERIMENTAL;
+
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.Optional;
+
+import org.apiguardian.api.API;
+import org.junit.platform.commons.logging.Logger;
+import org.junit.platform.commons.logging.LoggerFactory;
+
+/**
+ * {@code ClassOrderer} defines the API for ordering the <em>test classes</em>.
```

Comment:
We should document that it is only applied to top-level test classes, not nested ones.